browser/resources/settings/email_aliases_page/tests/email_aliases_list.test.tsx

```diff
@@ -0,0 +1,177 @@
+      domains: ['brave.com']
+    }
+  ]
+
+  const mockOnViewChange = jest.fn()
+  const mockOnListChange = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('renders the alias list', () => {
+    render(
+      <AliasList
+        aliases={mockAliases}
+        onViewChange={mockOnViewChange}
+        onListChange={mockOnListChange}
+        mappingService={mockMappingService}
+      />
+    )
+
+    // Check if alias information is displayed correctly
+    expect(screen.getByText(/test1@brave\.com/)).toBeInTheDocument()
+    expect(screen.getByText(/Test Alias 1/)).toBeInTheDocument()
+    expect(screen.getByText(/test2@brave\.com/)).toBeInTheDocument()
+    expect(screen.getByText(/Test Alias 2/)).toBeInTheDocument()
+  })
+
+  it('handles copy functionality', () => {
+    render(
+      <AliasList
+        aliases={mockAliases}
+        onViewChange={mockOnViewChange}
+        onListChange={mockOnListChange}
+        mappingService={mockMappingService}
+      />
+    )
+
+    // Click copy button
+    const copyButtons = screen.getAllByTestId('copy-toast')
+    fireEvent.click(copyButtons[0])
+
+    // Check if clipboard API was called
+    expect(navigator.clipboard.writeText)
+      .toHaveBeenCalledWith('test1@brave.com')
+  })
+
+  it('copies email when clicking alias label', () => {
+    render(
+      <AliasList
+        aliases={mockAliases}
+        onViewChange={mockOnViewChange}
+        onListChange={mockOnListChange}
+        mappingService={mockMappingService}
+      />
+    )
+
+    // Click the alias label
+    const aliasLabel = screen.getByText(/test1@brave\.com/)
+    fireEvent.click(aliasLabel)
+
+    // Check if clipboard API was called
+    expect(navigator.clipboard.writeText)
+      .toHaveBeenCalledWith('test1@brave.com')
+  })
+
+  it('disables create button when max aliases reached', () => {
+    const maxAliases = Array(10).fill(null).map((_, i) => ({
+      email: `test${i}@brave.com`,
+      note: `Test Alias ${i}`,
+      domains: ['brave.com']
+    }))
+
+    render(
+      <AliasList
+        aliases={maxAliases}
+        onViewChange={mockOnViewChange}
+        onListChange={mockOnListChange}
+        mappingService={mockMappingService}
+      />
+    )
+
+    // Check if create button is disabled
+    const createButton = screen.getByText(getLocale(
+      'emailAliasesCreateAliasLabel'))
+    expect(createButton.closest('leo-button'))
+      .toHaveAttribute('isdisabled', 'true')
+  })
+
+  it('handles delete functionality', async () => {
+    render(
+      <AliasList
+        aliases={mockAliases}
+        onViewChange={mockOnViewChange}
+        onListChange={mockOnListChange}
+        mappingService={mockMappingService}
+      />
+    )
+
+    // Click delete button
+    const deleteButtons = screen.getAllByText(getLocale('emailAliasesDelete'))
+    fireEvent.click(deleteButtons[0])
+
+    // Wait for the async operation to complete
+    await Promise.resolve()
```

Comment:
slightly unsure about this - presumably its because the mock for `deleteButton` is a promise so it doesn't get invoked till the next tick?

Reply:
removed now